kore/src/Kore/AST/Common.hs

```diff
@@ -1180,6 +1180,8 @@ data Pattern level domain variable child where
         :: !(Top level child) -> Pattern level domain variable child
     VariablePattern
         :: !(variable level) -> Pattern level domain variable child
+    SetVariablePattern
+        :: !(variable level) -> Pattern level domain variable child
```

Comment:
I would use a wrapper here:
```.hs
newtype SetVariable variable level = SetVariable (variable level)
```
and
```.hs
SetVariablePattern :: !(SetVariable variable level) -> Pattern level domain variable child
```
so that we can safely pass set variables around, but otherwise, yes this is what I had in mind.